packages/grafana-data/src/utils/url.ts

```diff
@@ -116,7 +116,11 @@ function getUrlSearchParams() {
       // key-value param
       const key = decodeURIComponent(keyValuePair[0]);
       const value = decodeURIComponent(keyValuePair[1]);
-      params[key] = value;
+      if (key in params) {
+        params[key].push(value);
+      } else {
+        params[key] = [value];
+      }
```

Comment:
Looks, good can we add tests for this? Thank you.

Reply:
It would be great if we could specify the return value for this function, should be `UrlQueryMap`. Also, the line 112 should be:
`const params: UrlQueryMap = {};` instead of `any`.

Reply:
Hi, thanks for reviewing! The tests are on their way, but I have a couple of concerns:

- I totally agree with specifying that the method returns a UrlQueryMap, but do we want params to be one too? Effectively it is going to comply with the type, but since here it is always going to be an array (as long as it is not a boolean), we avoid having to cast. If you'd rather have it as a UrlQueryMap then we would need to change line 120 to something like `params[key] = [...params[key] as any[], value];`. Whichever you prefer :)

- How much do we care about manually modified URLs? I believe the ones generated by Grafana are always going to work with the above code, but I can think of some manually written ones that could break it, like:
  - Defining a key-value parameter as a boolean parameter before it would cause it to break since true is not iterable.
  - Defining a key-value parameter as a boolean parameter after it would override all the previous values with true.

- A "solution" for the two problems above could be to change line 122 to `params[key] = [true];`. That would avoid any exceptions but, honestly, since the input is invalid I would rather throw one or validate the input in some other way if this issue is a concern.

Reply:
> Hi, thanks for reviewing! The tests are on their way, but I have a couple of concerns:
> 
> * I totally agree with specifying that the method returns a UrlQueryMap, but do we want params to be one too? Effectively it is going to comply with the type, but since here it is always going to be an array (as long as it is not a boolean), we avoid having to cast. If you'd rather have it as a UrlQueryMap then we would need to change line 120 to something like `params[key] = [...params[key] as any[], value];`. Whichever you prefer :)

Good points, I think we're avoiding `any` as much as possible because it leads to unclear types and code so I rather have the `unnecessary` casting and make baby steps into something clearer.

> * How much do we care about manually modified URLs? I believe the ones generated by Grafana are always going to work with the above code, but I can think of some manually written ones that could break it, like:
>   
>   * Defining a key-value parameter as a boolean parameter before it would cause it to break since true is not iterable.
>   * Defining a key-value parameter as a boolean parameter after it would override all the previous values with true.
> * A "solution" for the two problems above could be to change line 122 to `params[key] = [true];`. That would avoid any exceptions but, honestly, since the input is invalid I would rather throw one or validate the input in some other way if this issue is a concern.

I agree, let's keep the boolean as is for now and we can change that moving forward.